cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -4075,11 +4105,39 @@ macro(build_google_cloud_cpp_storage)
                         CURL::libcurl
                         Threads::Threads
                         OpenSSL::SSL
-                        OpenSSL::Crypto)
+                        OpenSSL::Crypto
+                        ZLIB::ZLIB)
   add_dependencies(google-cloud-cpp::storage google_cloud_cpp_ep)
 
-  list(APPEND ARROW_BUNDLED_STATIC_LIBS google-cloud-cpp::storage
+  list(APPEND
+       ARROW_BUNDLED_STATIC_LIBS
+       google-cloud-cpp::rest-internal
+       google-cloud-cpp::storage
```

This may not be a problem but could you put `google-cloud-cpp::rest-internal` after `google-cloud-cpp::storage` because `google-cloud-cpp::storage` depends on `google-cloud-cpp::rest-internal`?

```suggestion
       google-cloud-cpp::storage
       google-cloud-cpp::rest-internal
```